Sources/XMLCoder/Auxiliaries/XMLElement.swift

```diff
@@ -0,0 +1,280 @@
+//
+//  XMLElement.swift
+//  XMLCoder
+//
+//  Created by Vincent Esche on 12/18/18.
+//
+
+import Foundation
+
+internal class _XMLElement {
+    static let attributesKey = "___ATTRIBUTES"
+    static let escapedCharacterSet = [("&", "&amp"), ("<", "&lt;"), (">", "&gt;"), ("'", "&apos;"), ("\"", "&quot;")]
+    
+    var key: String
+    var value: String?
+    var attributes: [String: String] = [:]
+    var children: [String: [_XMLElement]] = [:]
+    
+    internal init(key: String, value: String? = nil, attributes: [String: String] = [:], children: [String: [_XMLElement]] = [:]) {
+        self.key = key
+        self.value = value
+        self.attributes = attributes
+        self.children = children
+    }
+    
+    static func createRootElement(rootKey: String, object: ArrayBox) -> _XMLElement? {
+        let element = _XMLElement(key: rootKey)
+        
+        _XMLElement.createElement(parentElement: element, key: rootKey, object: object)
+        
+        return element
+    }
+    
+    static func createRootElement(rootKey: String, object: DictionaryBox) -> _XMLElement? {
+        let element = _XMLElement(key: rootKey)
+        
+        _XMLElement.modifyElement(element: element, parentElement: nil, key: nil, object: object)
+        
+        return element
+    }
+    
+    fileprivate static func modifyElement(element: _XMLElement, parentElement: _XMLElement?, key: String?, object: DictionaryBox) {
+        let attributesBox = object[_XMLElement.attributesKey]?.dictionary
+        element.attributes = attributesBox?.unbox().mapValues { $0.xmlString } ?? [:]
+        
+        let objects = object.filter { key, _value in key != _XMLElement.attributesKey }
+        
+        for (key, box) in objects {
+            switch box {
```

Comment:
I'm not entirely convinced that this `switch` and duplication of `static func createElement(parentElement: _XMLElement, key: String, object:)` for every box type is the best solution. I'd rather prefer something like this:

```swift
protocol XMLBox {
  var xmlString: String { get }
}

// Every box type conforms to `Box` and implements `xmlString`

// ...
            static func createElement(parentElement: _XMLElement, key: String, object: Box) {
	        let element = _XMLElement(key: key, value: object.xmlString)
	        parentElement.children[key] = (parentElement.children[key] ?? []) + [element]
	    }
//...
```

This `switch` would go away then as well.